### PR TITLE
Implement XML responses for SNS (for Boto3)

### PR DESCRIPTION
**Pull request by pior**
Hi @spulec,

I partially implemented the XML responses for the SNS backend (the topic related responses).
There is a new Boto3 specific test: https://github.com/ludia/moto/blob/feature/implement-sns-xml-response/tests/test_sns/test_topics_boto3.py 

If you think this is the right direction, I could probably finish this today.

### CODE REVIEW

File: moto/sns/models.py [hunk 1]
```diff
@@ -284,9 +285,9 @@ def set_endpoint_attributes(self, arn, attributes):
             }
         }
     }]
-}
+})
```
Comment (pior) [inline moto/sns/models.py]: Even in the JSON representation, the *_policy are represented as string (not as nested structure)

File: moto/sns/responses.py [hunk 2]
```diff
@@ -335,10 +411,101 @@ def set_endpoint_attributes(self):
...
+                    }
                 }
-            }
-        })
+            })
+
+        template = self.response_template(_TEMPLATE)
+        return template.render()
+
+_TEMPLATE = ""
```
Comment (pior) [inline moto/sns/responses.py]: This will go away.

**Comment by spulec** [comment]
This is great!

I'm happy to merge now, but can wait if you want to keep working on the rest on this branch.

**Comment by pior** [comment]
@spulec Great! 

I'll just finish the complete implementation of SNS.

**Comment by pior** [comment]
@spulec Done!

**Comment by spulec** [comment]
Awesome! Thanks for all of your work.

**Pull request merged**